src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_LENGTH = 50;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private static final int FIGURE_NUMBER = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure [] figure = new Figure[FIGURE_NUMBER];
+        figure[0] = new Square(random.nextInt(MAX_LENGTH), colorSupplier.getRandomColor());
+        figure[1] = new Circle(random.nextInt(MAX_LENGTH), colorSupplier.getRandomColor());
+        figure[2] = new IsoscelesTrapezoid(random.nextInt(MAX_LENGTH), random.nextInt(MAX_LENGTH),
+                random.nextInt(MAX_LENGTH), colorSupplier.getRandomColor());
+        figure[3] = new Rectangle(random.nextInt(MAX_LENGTH),
+                random.nextInt(MAX_LENGTH), colorSupplier.getRandomColor());
+        figure[4] = new RightTriangle(random.nextInt(MAX_LENGTH),
+                random.nextInt(MAX_LENGTH), colorSupplier.getRandomColor());
+        return figure[random.nextInt(figure.length)];

```

Comment:
lets create a switch-case. Because you are making too many objects each time this method is invoked when you need just 1.
Also lets create private methods to get each random figure - this would make your code easier to read
```
    swith(randomNumber) {
         case 0: return getRandomSquare();
         ....
    }
```